Empire/Empire.cpp

```diff
@@ -33,51 +33,55 @@ namespace {
-                    float new_front_factor = std::atof(UserString("FUNCTIONAL_PRODUCTION_QUEUE_FRONTLOAD_FACTOR").c_str());
-                    if (new_front_factor > 0.0f && new_front_factor <= 0.3f)
-                        front_load_factor = new_front_factor;
-                }
-            } catch (...) {}
-        }
-        return front_load_factor;
-    }
-
-    float GetQueueToppingFactor() {
-        static float topping_up_factor = -1.0;
-        if (topping_up_factor == -1.0) {
-            topping_up_factor = 0.0;
-            try {
-                if (UserStringExists("FUNCTIONAL_PRODUCTION_QUEUE_FRONTLOAD_FACTOR")) {
-                    float new_front_factor = std::atof(UserString("FUNCTIONAL_PRODUCTION_QUEUE_FRONTLOAD_FACTOR").c_str());
-                    if (new_front_factor > 0.0f && new_front_factor <= 0.3f)
-                        topping_up_factor = new_front_factor;
-                }
-            } catch (...) {}
-        }
-        return topping_up_factor;
-    }
-
-// FUNCTIONAL_PRODUCTION_QUEUE_FRONTLOAD_FACTOR and FUNCTIONAL_PRODUCTION_QUEUE_TOPPING_UP_FACTOR specify global_settings.txt values that affect how the ProductionQueue will limit
-// allocation towards building a given item on a given turn.  The base amount of maximum allocation per turn (if the player has enough PP available) is the item's total
-// cost, divided over its minimum build time.  Sometimes complications arise, though, which unexpectedly delay the completion even if the item had been fully-funded 
-// every turn, because costs have risen partway through (such as due to increasing ship costs resulting from recent ship constructoin completion and ensuing increase 
-// of Fleet Maintenance costs.  These two settings provide a mechanism for some allocation leeway to deal with mid-build cost increases without causing the project 
-// completion to take an extra turn because of the small bit of increased cost.  The settings differ in the timing of the extra allocation allowed.
-// Both factors have a minimum value of 0.0 and a maximum value of 0.3.
-
-// Making the frontloaded factor greater than zero increases the per-turn allocation cap by the specified percentage (so it always spreads the extra allocation across all turns).
-// Making the topping-up option nonzero allows the final turn allocation cap to be increased by the specified percentage of the total cost, if needed (and then subject to 
-// availability of course). They can both be nonzero, although to avoid that introducing too much interaction complexity into the minimum build time safeguard for topping-up, 
-// the topping-up percentage will be reduced by the frontloading setting. 
-
-// Note that for very small values of the options (less than 5%), when dealing with very low cost items the effect/protection may be noticeably less than expected because of
-// interactions with the ProductionQueue Epsilon value (0.01)
+    void AddOptions(OptionsDB& db) {
```

Comment:
These are gameplay mechanics modifications, and thus not appropriate to put in the OptionsDB.